### PR TITLE
Fixes #29331 - use patternfly v4 fonts

### PR DESCRIPTION
**Pull request by sharvit**
Use pf4 fonts (RedhatText) and link colors (blue).

# Examples

screenshot-centos7-luna-devel sharvit-fedorabook-t480s example com-2020 03 15-14_10_42

screenshot-centos7-luna-devel sharvit-fedorabook-t480s example com-2020 03 15-14_15_58 (1)

screenshot-centos7-luna-devel sharvit-fedorabook-t480s example com-2020 03 15-14_15_58

### CODE REVIEW

File: app/assets/stylesheets/navigation.scss [hunk 1]
```diff
@@ -44,6 +44,10 @@
   span {
```
Comment (amirfefer) [inline app/assets/stylesheets/navigation.scss]: does it make sense moving navigation styles to webpack?

**Review by amirfefer — COMMENTED** [review]
Overall I'm :+1: with this change, though it isn't dramatically noticeable, it's a good strart

File: app/assets/stylesheets/navigation.scss (hunk 1 above)
Comment (sharvit) [reply app/assets/stylesheets/navigation.scss]: No, we aim to replace the navigation component later.
Anyway, not in this PR please.

File: package.json [hunk 2]
```diff
@@ -20,7 +20,7 @@
     "create-react-component": "yo react-domain"
   },
   "dependencies": {
-    "@theforeman/vendor": "^4.0.7",
+    "@theforeman/vendor": "^4.2.0",
```
Comment (amirfefer) [inline package.json]: do we need packaging ack here?

File: package.json [hunk 3]
```diff
@@ -30,11 +30,11 @@
...
-    "@theforeman/builder": "^4.0.7",
-    "@theforeman/eslint-plugin-foreman": "^4.0.7",
-    "@theforeman/stories": "^4.0.7",
-    "@theforeman/test": "^4.0.7",
-    "@theforeman/vendor-dev": "^4.0.7",
+    "@theforeman/builder": "^4.2.0",
+    "@theforeman/eslint-plugin-foreman": "^4.2.0",
+    "@theforeman/stories": "^4.2.0",
+    "@theforeman/test": "^4.2.0",
+    "@theforeman/vendor-dev": "^4.2.0",
```
Comment (johnpmitsch) [inline package.json]: @sharvit I opened up https://github.com/theforeman/foreman/pull/7590 - but I didn't see the versions were updated here. Should these be `^5.0.0` now that that is released?

**Comment by johnpmitsch** [comment]
@sharvit  Thanks for the hard work getting pf4 into Foreman. The package bumps and fonts are needed for https://github.com/Katello/katello/pull/8644

Is there anything I can do to help move this PR along?

**Comment by sharvit** [comment]
Thanks @johnpmitsch and @amirfefer.

I ended up doing a reset to the commits in v5.0.0 and deprecated that version.
Sorry for the confusion about it.

packaging-pr:
https://github.com/theforeman/foreman-packaging/pull/5106

**Review by tbrisker — COMMENTED** [review]
The secondary menu item font size seems very small with this change.
New hosts dashboard widget overflows if there are enough new hosts, i assume the same is true of other widgets:
image

**Comment by sharvit** [comment]
Thanks @tbrisker, I changed the base font-size from `--pf-global--FontSize--sm` to `--pf-global--FontSize--xs` while the recommended is `--pf-global--FontSize--md`.

Now it should look more similar to what we had before so those layout issues are now fixed. From the other side, I see it as a technical debt because pf4 encourages using bigger fonts and fewer components in one screen so the user would have more focus.

IMO we can start with the `xs` font size and over time increase it to `md`.
We can set the font to `sm` after branching and try to address layout issues until the next branching.

**Comment by tbrisker** [comment]
> Thanks @tbrisker, I changed the base font-size from `--pf-global--FontSize--sm` to `--pf-global--FontSize--xs` while the recommended is `--pf-global--FontSize--md`.
> 
> Now it should look more similar to what we had before so those layout issues are now fixed. From the other side, I see it as a technical debt because pf4 encourages using bigger fonts and fewer components in one screen so the user would have more focus.
> 
> IMO we can start with the `xs` font size and over time increase it to `md`.
> We can set the font to `sm` after branching and try to address layout issues until the next branching.

Looks like at least part of the overflow is due to the header breaking up (operating system), would it make sense to modify column widths instead so it all fits and/or reduce the number of lines displayed in the widget?

**Comment by tbrisker** [comment]
Also, packaging has been merged, this needs a rebase

**Comment by sharvit** [comment]
> Looks like at least part of the overflow is due to the header breaking up (operating system), would it make sense to modify column widths instead so it all fits and/or reduce the number of lines displayed in the widget?

Yes, modified the os column and set the tables to have pf4 line-height and it looks fine with the `sm` font size.

**Comment by tbrisker** [comment]
Looks like there's still some font sizes that are off, mostly the navbar menus are very small but also the action buttons seem a bit smaller than the rest of the row:
Screenshot from 2020-05-26 15-52-20

**Comment by sharvit** [comment]
Thanks @tbrisker, I did some small adjustments so it would be similar to what we have now.

**Comment by tbrisker** [comment]
Please rebase @sharvit

**Comment by sharvit** [comment]
Rebased

**Review by tbrisker — APPROVED** [review]
Tested, looks good! The menu items are still a tad bit small, but let's fix that and any additional issues in following PRs.
Thanks @sharvit !

**Pull request merged**